add component module

ameerhamzakhalid opened the pull request:
- I added React components.
- I added React props.
- I added React state.

ameerhamzakhalid committed:
add component module

I added React components.
I added React props.
I added React state.

addod19 commented on src/App.js:
```diff
@@ -1,27 +1,10 @@
+import Calculator from './components/Calculator';
 import './App.css';
 
 function App() {
```
- [ ] It would be better if we convert our functional component into class based component
![Screenshot from 2023-01-12 12-45-46](https://user-images.githubusercontent.com/46388113/212069439-e9dfda00-cdeb-4826-9e85-3ddccefdd951.png)

addod19 reviewed (changes requested):
### Hi @ameerhamzakhalid, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Git-flow followed :100: 
- [ ] Design matches up with requirements :1st_place_medal: 
- [ ] `Components` directory present :+1: 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Capitalizing our commit messages :rocket: 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

ameerhamzakhalid committed:
add class

I converted the functional component into class based component, as suggested by code reviewer in previous review.

ameerhamzakhalid committed:
fixed linter

I fixed linter errors

CollinsTatang reviewed (approved):
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
![image](https://user-images.githubusercontent.com/25227022/165483662-7b06c812-4cc0-40b9-a84c-4729d4cf10c3.gif)



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
